cmd/generate.go

```diff
@@ -0,0 +1,171 @@
+			service.Ports = append(service.Ports, dockerPort)
+
+			//set container env vars for healthcheck, and port
+			//so that apps can simulate running in harbor
+			service.Environment["PORT"] = strconv.Itoa(port.Value)
+			service.Environment["HEALTHCHECK"] = port.Healthcheck
+		}
+
+		//copy shipment, environment, provider level env vars down to the
+		//container level so that they can be used in docker-compose
+
+		//shipment
+		copyEnvVars(shipmentObject.ParentShipment.EnvVars, service.Environment, nil)
+
+		//environment
+		copyEnvVars(shipmentObject.EnvVars, service.Environment, nil)
+
+		//todo: provider
+
+		//container
+		copyEnvVars(container.EnvVars, service.Environment, nil)
+
+		//add service to list
+		dockerCompose.Services[container.Name] = service
+	}
+
+	//prompt if the file already exist
+	if _, err := os.Stat(DockerComposeFile); err == nil {
+		//exists
+		fmt.Print("docker-compose.yml already exists. Overwrite? ")
+		if askForConfirmation() {
+			SerializeDockerCompose(dockerCompose, DockerComposeFile)
+		}
+	} else {
+		//doesn't exist
+		SerializeDockerCompose(dockerCompose, DockerComposeFile)
+	}
+
+	//now generate harbor-compose.yml
+
+	//convert a Shipment object into a HarborCompose object with a single shipment
+	harborCompose := HarborCompose{
+		Shipments: make(map[string]ComposeShipment),
+	}
+
+	composeShipment := ComposeShipment{
+		Env:         shipmentObject.Name,
+		Group:       shipmentObject.ParentShipment.Group,
+		Environment: make(map[string]string),
+	}
+
+	//populate env vars
+
+	//track special envvars
+	special := map[string]string{}
+
+	//shipment
+	copyEnvVars(shipmentObject.ParentShipment.EnvVars, composeShipment.Environment, special)
+
+	//environment
+	copyEnvVars(shipmentObject.EnvVars, composeShipment.Environment, special)
+
+	//todo: provider
+
+	//now populate other harbor-compose metadata
+	composeShipment.Barge = special["BARGE"]
+	composeShipment.Product = special["PRODUCT"]
+	composeShipment.Project = special["PROJECT"]
+	composeShipment.Property = special["PROPERTY"]
+
+	//set replicas from the ec2 provider
+	composeShipment.Replicas = ec2Provider(shipmentObject.Providers).Replicas
+
+	//add containers
+	for container := range dockerCompose.Services {
+		composeShipment.Containers = append(composeShipment.Containers, container)
+	}
+
+	//add single shipment to list
+	harborCompose.Shipments[shipmentObject.ParentShipment.Name] = composeShipment
+
+	//prompt if the file already exist
+	if _, err := os.Stat(HarborComposeFile); err == nil {
+		//exists
+		fmt.Print("harbor-compose.yml already exists. Overwrite? ")
+		if askForConfirmation() {
+			SerializeHarborCompose(harborCompose, HarborComposeFile)
+			fmt.Println("done")
+		}
+	} else {
+		//doesn't exist
+		SerializeHarborCompose(harborCompose, HarborComposeFile)
+		fmt.Println("done")
+	}
+
+}
+
+//find the ec2 provider
+func ec2Provider(providers []ProviderPayload) *ProviderPayload {
```

Comment:
we need to rethink a better way to implement the providers. If we add a new one in the future, this would break. It's ok for now, just noting.